test/gtest/ucp/test_ucp_device.cc

```diff
@@ -24,6 +26,64 @@ class test_ucp_device : public ucp_test {
             receiver().connect(&sender(), get_ep_params());
         }
     }
+
+    class mem_list {
+        using mapped_buffer_ptr = std::unique_ptr<mapped_buffer>;
+
+        struct params {
+            std::unique_ptr<ucp_mem_list_elem_t[]> elems;
+            std::vector<ucs::handle<ucp_rkey_h>>   rkeys;
+            ucp_mem_list_params_t                  params;
+        };
+
+        entity&                        m_sender;
+        entity&                        m_receiver;
+        std::vector<mapped_buffer_ptr> m_src;
+        std::vector<mapped_buffer_ptr> m_dst;
+        params                         m_params;
+
+    public:
+        mem_list(entity &sender, entity &receiver) :
+            m_sender(sender), m_receiver(receiver)
+        {
+        }
+
+        void add(size_t size, unsigned count = 1,
+                 ucs_memory_type_t mem_type = UCS_MEMORY_TYPE_CUDA)
+        {
+            for (auto i = 0; i < count; ++i) {
+                m_src.emplace_back(
+                        new mapped_buffer(size, m_sender, 0, mem_type));
+                m_dst.emplace_back(
+                        new mapped_buffer(size, m_receiver, 0, mem_type));
+            }
+        }
+
+        const ucp_mem_list_params_t &
```

Comment:
do we really need class? can make single function that builds mem_list according to some criteria?